cli/cmd/tapRunner.go

```diff
@@ -598,6 +611,96 @@ func watchApiServerPod(ctx context.Context, kubernetesProvider *kubernetes.Provi
 	}
 }
 
+func watchTapperPod(ctx context.Context, kubernetesProvider *kubernetes.Provider, cancel context.CancelFunc) {
+	podExactRegex := regexp.MustCompile(fmt.Sprintf("^%s.*", mizu.TapperDaemonSetName))
+	added, modified, removed, errorChan := kubernetes.FilteredWatch(ctx, kubernetesProvider, []string{config.Config.MizuResourcesNamespace}, podExactRegex)
+	var prevPodPhase core.PodPhase
+	for {
```

Comment:
i think we should do here same as we did in watchApiServerPod, if not all tappers are up and running after 25 seconds we should cancel and close the application.
WDYT @gadotroee ?

Reply:
I think we should print "Mizu is available at..." after all Mizu components are up and running which are APIServer and all tappers

Reply:
@IgorGov once the API server pod is ready, the app is ready so it's opened in browser. Then the events related to tapper pod is streamed to the user via WebSocket in the form of toast messages.

If there are insufficient resources for the tapper pod then the deployment is canceled. If the tapper pod is terminated then the user is informed through toast messages. Such that they can restart Mizu with more resources.

The only two case I see here that might cause the termination of the tapper pod are:
- OOMKilled.
- A runtime error.

Other than these we can assume that it's guaranteed to be deployed if there are enough resources in the cluster.

Reply:
I mean termination case can only happen after the deployment so it will be deployed if there are enough resources.

Reply:
the issue is that tappers are updates after initialization
the can be (and probably will be) restarted more than one time if there is change in the pods that matching the regex (because the nodes that we need to deploy the daemon set on can be changed).

I didn't take a deeper look on the PR so I will try to do it and see if I have comments

Reply:
@gadotroee https://github.com/up9inc/mizu/pull/304/commits/4d3f0f7ad0959ed55cf7c0aadf7cae19189a3bd2 fixes the issues related to tapper updates. (removal and recreation)